### PR TITLE
order_detail_management

### PR DESCRIPTION
**Pull request by VanHoa18042000**
## Related Tickets
- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/51563)

## WHAT (optional)
- show list order, action order_detail

## HOW
- Add css, js for home page , admin , layout login . And add gem i18n

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
Screenshot from 2022-08-23 11-16-13

### CODE REVIEW

**Comment by VanHoa18042000** [comment]
ready

File: app/controllers/admin/order_details_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Admin::OrderDetailsController < ApplicationController
+  # before_action :logged_in_user
```
Comment (huongnt-2545) [inline app/controllers/admin/order_details_controller.rb:2]: ko dùng thì xóa đi em

File: app/views/admin/order_details/_index.html.erb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<div class="main-body">
+  <div class="page-wrapper">
+    <div class="row">
+      <div class="col-xl-12">
+        <div class="card">
+          <div class="card-header">
+            <div class="box-detail p-0">
+              <div class="box-detail__item">
+                <strong><%= t "admin.orders.name_user_order" %>: </strong>
+                <span><%= @order.user.name %></span>
```
Comment (huongnt-2545) [inline app/views/admin/order_details/_index.html.erb:10]: dùng delegate

File: app/views/admin/order_details/_index.html.erb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                <span><%= show_datetime(@order.date_return) %></span>
+              </div>
+            </div>
+          </div>
+          <div class="card-body table-border-style">
+            <div class="table-responsive">
+              <table class="table table-hover">
+                <thead>
+                  <tr>
+                    <th style="width: 5%;">
```
Comment (huongnt-2545) [inline app/views/admin/order_details/_index.html.erb:31]: ko style trực tiếp trong file view như này em nhé
-> move vào file css

File: app/views/admin/order_details/update.js.erb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    $("#action-order-detail-<%= order_detail.id %> .status").removeClass("active")
+
+
+    if("<%= order_detail.status %>" != "pending"){
+      $("#action-order-detail-<%= order_detail.id %> .<%= order_detail.status %>-status").addClass("active")
+      $(".quantity_real-of-order-detail-<%= order_detail.id %>").hide()
+      $(".box-status-order-detail#action-order-detail-<%= order_detail.id %>").html("<div class='badge badge-no-acion mt-3 bg-secondary'><%= t 'no_action' %></div>")
+    }
+    if("<%= order_detail.status %>" == "accept"){
+      $("#order_note_admin").val($("#order_note_admin").val() + "\nCuốn sách <%= order_detail.book.name %> của bạn mượn được chấp nhận với số lượng là <%= order_detail.quantity_real %>")
```
Comment (huongnt-2545) [inline app/views/admin/order_details/update.js.erb:14]: I18n text
fix tương tự bên dưới

File: app/views/admin/orders/_order.html.erb [hunk 5]
```diff
@@ -0,0 +1,39 @@
+<tr>
+  <td>
+    <%= order_counter + 1 %>
+  </td>
+  <td>
+    <%= order.user.name %>
```
Comment (huongnt-2545) [inline app/views/admin/orders/_order.html.erb:6]: dùng delegate

**Comment by VanHoa18042000** [comment]
ready

**Review by huongnt-2545 — COMMENTED** [review]
cmt
vẫn còn comment chưa fix em nhé

**Comment by VanHoa18042000** [comment]
ready

**Review by anlvq-1058 — COMMENTED** [review]
merged

**Pull request merged**